lib_eio_posix/eio_posix_stubs.c

```diff
@@ -27,8 +27,10 @@
 
 #ifdef ARCH_SIXTYFOUR
 #define Int63_val(v) Long_val(v)
+#define caml_copy_int63(v) Val_long(v)
 #else
 #define Int63_val(v) (Int64_val(v)) >> 1
+#define caml_copy_int63 caml_copy_int64
```

Comment:
Int63 uses the *top* 63-bits (this was a mistake in my branch):
```suggestion
#define caml_copy_int63(v) caml_copy_int64(v << 1)
```